Shoko.Server/API/v3/Controllers/SeriesController.cs

```diff
@@ -134,6 +134,26 @@ public ActionResult<List<SeriesRelation>> GetShokoRelationsBySeriesID([FromRoute
                 .ToList();
         }
         
+        /// <summary>
+        /// Get a paginated list of <see cref="Series"/> without local files, available to the current <see cref="User"/>.
+        /// </summary>
+        /// <param name="pageSize">The page size.</param>
+        /// <param name="page">The page index.</param>
+        /// <returns></returns>
+        [HttpGet("WithoutFiles")]
+        public ActionResult<ListResult<Series>> GetSeriesWithoutFiles([FromQuery] [Range(0, 100)] int pageSize = 50, [FromQuery] [Range(1, int.MaxValue)] int page = 1)
+        {
+            var user = User;
+            return RepoFactory.AnimeSeries.GetAll()
+                .Select(series => (series, seriesName: series.GetSeriesName().ToLowerInvariant()))
+                .Where(tuple => {
+                    var (series, seriesName) = tuple;
+                    return user.AllowedSeries(series) && series.GetAnimeEpisodesCountWithVideoLocal() == 0;
+                })
+                .OrderBy(a => a.seriesName)

```

Comment:
We don't need to get the series name early if we're only going to use it for ordering the result.

Reply:
I'm too sick to figure out how to do multi-line suggestions, so here is what I would've done since the series name is not needed before the `.OrderBy`;
```cs
            return RepoFactory.AnimeSeries.GetAll()
                .Where(series => user.AllowedSeries(series) && series.GetAnimeEpisodesCountWithVideoLocal() == 0)
                .OrderBy(series => series.GetSeriesName().ToLowerInvariant())
                .ToListResult(tuple => new Series(HttpContext, tuple.series), page, pageSize);
```

Reply:
Done